# FIFA 19 — Insights

### 1. Carregando o dataset

In [69]:
import pandas as pd

df = pd.read_csv('fifa.csv', index_col=0)
df.head()

,ID,Name,Age,Photo,Nationality,Flag,Overall,Potential,Club,Club Logo,...,Composure,Marking,StandingTackle,SlidingTackle,GKDiving,GKHandling,GKKicking,GKPositioning,GKReflexes,Release Clause
0,158023,L. Messi,31,https://cdn.sofifa.org/players/4/19/158023.png,Argentina,https://cdn.sofifa.org/flags/52.png,94,94,FC Barcelona,https://cdn.sofifa.org/teams/2/light/241.png,...,96.0,33.0,28.0,26.0,6.0,11.0,15.0,14.0,8.0,€226.5M
1,20801,Cristiano Ronaldo,33,https://cdn.sofifa.org/players/4/19/20801.png,Portugal,https://cdn.sofifa.org/flags/38.png,94,94,Juventus,https://cdn.sofifa.org/teams/2/light/45.png,...,95.0,28.0,31.0,23.0,7.0,11.0,15.0,14.0,11.0,€127.1M
2,190871,Neymar Jr,26,https://cdn.sofifa.org/players/4/19/190871.png,Brazil,https://cdn.sofifa.org/flags/54.png,92,93,Paris Saint-Germain,https://cdn.sofifa.org/teams/2/light/73.png,...,94.0,27.0,24.0,33.0,9.0,9.0,15.0,15.0,11.0,€228.1M
3,193080,De Gea,27,https://cdn.sofifa.org/players/4/19/193080.png,Spain,https://cdn.sofifa.org/flags/45.png,91,93,Manchester United,https://cdn.sofifa.org/teams/2/light/11.png,...,68.0,15.0,21.0,13.0,90.0,85.0,87.0,88.0,94.0,€138.6M
4,192985,K. De Bruyne,27,https://cdn.sofifa.org/players/4/19/192985.png,Belgium,https://cdn.sofifa.org/flags/7.png,91,92,Manchester City,https://cdn.sofifa.org/teams/2/light/10.png,...,88.0,68.0,58.0,51.0,15.0,13.0,5.0,10.0,13.0,€196.4M


In [70]:
df.columns

Index(['ID', 'Name', 'Age', 'Photo', 'Nationality', 'Flag', 'Overall',
       'Potential', 'Club', 'Club Logo', 'Value', 'Wage', 'Special',
       'Preferred Foot', 'International Reputation', 'Weak Foot',
       'Skill Moves', 'Work Rate', 'Body Type', 'Real Face', 'Position',
       'Jersey Number', 'Joined', 'Loaned From', 'Contract Valid Until',
       'Height', 'Weight', 'LS', 'ST', 'RS', 'LW', 'LF', 'CF', 'RF', 'RW',
       'LAM', 'CAM', 'RAM', 'LM', 'LCM', 'CM', 'RCM', 'RM', 'LWB', 'LDM',
       'CDM', 'RDM', 'RWB', 'LB', 'LCB', 'CB', 'RCB', 'RB', 'Crossing',
       'Finishing', 'HeadingAccuracy', 'ShortPassing', 'Volleys', 'Dribbling',
       'Curve', 'FKAccuracy', 'LongPassing', 'BallControl', 'Acceleration',
       'SprintSpeed', 'Agility', 'Reactions', 'Balance', 'ShotPower',
       'Jumping', 'Stamina', 'Strength', 'LongShots', 'Aggression',
       'Interceptions', 'Positioning', 'Vision', 'Penalties', 'Composure',
       'Marking', 'StandingTackle', 'SlidingTackle', 'GKDiv

### 2. Métricas gerais

In [71]:
metricas = df[['Age','Overall', 'Potential']]

In [72]:
metricas.describe()

,Age,Overall,Potential
count,18207.000000,18207.000000,18207.000000
mean,25.122206,66.238699,71.307299
std,4.669943,6.908930,6.136496
min,16.000000,46.000000,48.000000
25%,21.000000,62.000000,67.000000
50%,25.000000,66.000000,71.000000
75%,28.000000,71.000000,75.000000
max,45.000000,94.000000,95.000000


###3. Contagem

In [73]:
count_players_by_nat = df['Nationality'].value_counts().head(10)
count_players_by_nat

,count
Nationality,
England,1662
Germany,1198
Spain,1072
Argentina,937
France,914
Brazil,827
Italy,702
Colombia,618
Japan,478


In [74]:
count_preferred_foot = df['Preferred Foot'].value_counts()
count_preferred_foot

,count
Preferred Foot,
Right,13948
Left,4211


### 4.Insights

In [75]:
def limpar_valor(valor_texto):
    if isinstance(valor_texto, str):
        # 1.Remove o euro e espaços
        valor_texto = valor_texto.replace('€', '').replace(' ', '')

        # 2.Se for Milhão (M)
        if 'M' in valor_texto:
            valor_texto = valor_texto.replace('M', '')
            return float(valor_texto) * 1000000

        # 3.Se for Mil (K)
        elif 'K' in valor_texto:
            valor_texto = valor_texto.replace('K', '')
            return float(valor_texto) * 1000

    return 0.0 #Caso o valor seja vazio ou 0


#Aplicando na coluna Value
df['Value_Num'] = df['Value'].apply(limpar_valor)

#Aplicando na coluna Wage
df['Wage_Num'] = df['Wage'].apply(limpar_valor)

#Conferindo o resultado
print(df[['Name', 'Value', 'Value_Num']].head())

                Name    Value    Value_Num
0           L. Messi  €110.5M  110500000.0
1  Cristiano Ronaldo     €77M   77000000.0
2          Neymar Jr  €118.5M  118500000.0
3             De Gea     €72M   72000000.0
4       K. De Bruyne    €102M  102000000.0


In [76]:
value_players = df[['Name', 'Age', 'Value_Num','Wage_Num']]
value_players.sort_values('Value_Num', ascending =False)

,Name,Age,Value_Num,Wage_Num
2,Neymar Jr,26,118500000.0,290000.0
0,L. Messi,31,110500000.0,565000.0
4,K. De Bruyne,27,102000000.0,355000.0
5,E. Hazard,27,93000000.0,340000.0
15,P. Dybala,24,89000000.0,205000.0
...,...,...,...,...
2724,C. Keşerü,31,0.0,0.0
11506,R. Quiñonez,26,0.0,0.0
16135,S. Bose,22,0.0,0.0
16265,R. Borges,26,0.0,0.0


In [77]:
value_players.sort_values('Wage_Num', ascending =False)

,Name,Age,Value_Num,Wage_Num
0,L. Messi,31,110500000.0,565000.0
7,L. Suárez,31,80000000.0,455000.0
6,L. Modrić,32,67000000.0,420000.0
1,Cristiano Ronaldo,33,77000000.0,405000.0
8,Sergio Ramos,32,51000000.0,380000.0
...,...,...,...,...
12811,G. Singh Sandhu,26,0.0,0.0
12930,K. Despodov,21,0.0,0.0
13185,M. Raynov,26,0.0,0.0
13096,N. Boxall,26,0.0,0.0


### 5.Agrupamento

In [78]:
age_by_club = df.groupby('Club')['Age'].mean()
age_by_club.sort_values().head(5)

,Age
Club,
FC Nordsjælland,20.259259
FC Groningen,21.423077
Bohemian FC,21.480000
FC Sochaux-Montbéliard,21.678571
FC Admira Wacker Mödling,21.925926


In [79]:
overall_by_club = df.groupby('Club')['Overall'].mean()
overall_by_club.sort_values(ascending=False).head(5)

,Overall
Club,
Juventus,82.280000
Napoli,80.000000
Inter,79.750000
Real Madrid,78.242424
Milan,78.074074


### Conclusões

* O dataset conta com **18.207** jogadores, com média de **25** anos de idade, **66** de overall e **71** de potencial. A Inglaterra domina em quantidade, com **1.662** jogadores registrados, mais que o dobro do Brasil, que aparece em **6º** com **827**.

* Em relação ao pé preferido, a maioria esmagadora dos jogadores usa o **pé direito (13.948)**, contra apenas **4.211** canhotos, uma proporção de aproximadamente **77%** para **23%**.

* O jogador mais valioso do dataset é **Neymar Jr (PSG)**, avaliado em **€118,5** milhões com 26 anos e 92 de overall. Já o maior salário é de**L. Messi**, recebendo **€565** mil por semana pelo FC Barcelona.

* Por fim, a Juventus é o clube com o maior overall médio do jogo, com 82.3, seguida de Napoli (80.0) e Inter (79.8). O clube mais jovem em média é o **FC Nordsjælland** da Dinamarca, com apenas **20.3** anos de média de elenco.